src/shared/dissect-image.c

```diff
@@ -1661,6 +1701,40 @@ int dissect_image_and_warn(
         }
 }
 
+bool dissected_image_can_do_verity(const DissectedImage *image, unsigned partition_number) {
+        int k;
+
+        k = PARTITION_VERITY_OF(partition_number);
+        if (k >= 0)
+                return true;
+        if (image->can_verity) {
+                /* image->can_verity is meaningful on a per-partition check only with a single file system */
+                for (unsigned i = PARTITION_ROOT + 1; i < _PARTITION_DESIGNATOR_MAX; i++)
+                        if (image->partitions[i].found)
+                                return false;
+                return true;
+        }
+
+        return false;
+}
+
+bool dissected_image_has_verity(const DissectedImage *image, unsigned partition_number) {
+        int k;
+
+        k = PARTITION_VERITY_OF(partition_number);
+        if (k >= 0 && image->partitions[k].found)
+                return true;
+        if (image->verity) {
+                /* image->verity is meaningful on a per-partition check only with a single file system */
+                for (unsigned i = PARTITION_ROOT + 1; i < _PARTITION_DESIGNATOR_MAX; i++)
+                        if (image->partitions[i].found)
+                                return false;
+                return true;
+        }
```

Comment:
with the `naked_file_system` boolean suggested above this would become:

```c
if (image->naked_file_system)
        return designator == PARTITION_ROOT && image->verity;

k = PARTITION_VERITY_OF(designator);
return k >= 0 && image->partitions[k].found;
```

Reply:
Done